internal/directives/argocd_updater.go

```diff
@@ -536,6 +536,15 @@ func (a *argocdUpdater) syncApplication(
 	for _, source := range app.Spec.Sources {
 		app.Operation.Sync.Revisions = append(app.Operation.Sync.Revisions, source.TargetRevision)
 	}
+	// TODO(krancour): This is a workaround for the Argo CD Application controller
+	// not handling this correctly itself. It is Argo CD's API server that usually
+	// handles this, but we are bypassing the API server here.
+	//
+	// See issue: https://github.com/argoproj/argo-cd/issues/20875
+	//
+	// We can remove this hack once the issue is resolved and all Argo CD versions
+	// without the fix have reached their EOL.
+	app.Status.OperationState = nil
```

Comment:
@krancour, shall we also do
```
dst.Object["status"] = src.Object["status"]
```
below?

Reply:
Ugh. Missed that, although it probably shouldn't be status, but _just_ status.operationState. As previously discussed, we really don't want to be messing with status at all, so we need to be very surgical about what we patch here.

Reply:
Was there a fix done for this?

Reply:
Look down 👇